server/controllers/user.controller.js

```diff
@@ -60,40 +61,37 @@ export const signup = async (req, res) => {
 };
 
 
-export const signin = (req, res) => {
-//   if (!req.body.email || !req.body.password)
-//     return res.status(400).send({
-//       status: 400,
-//       error: "please check your credentials"
-//     });
-
-//   const newUser = User.find(c => c.email == req.body.email);
-
-//   if (!newUser)
-//     return res.status(400).send({
-//       status: 400,
-//       error: "User with provided email or password do not exist!"
-//     });
-
-//   const token = jwt.sign({ email: newUser.email }, process.env.MY_SECRET, {
-//     expiresIn: "2d"
-//   });
+export const signin = async (req, res) => {
+    let { email, password } = req.body;
+    
+    const text = 'SELECT * FROM users WHERE email = $1';
+    try {
+      const { rows } = await pool.query( text, [ email ] );
+      
+      if (rows.length===0 && !decryptpw(rows[0].password,password)) {
+        return res.status(401).send({'message': ' Incorrect password or password' });
+      }
+  
 
-//   const id = uuid.v1();
-//   if (newUser.password == req.body.password) {
-//     const { firstname, lastname, email } = newUser;
-//     return res.status(201).send({
-//       status: 201,
-//       message: "successfully log in",
-//       data: {
-//         token
-//       },
-//       userDetails: {
-//         id,
-//         firstname,
-//         lastname,
-//         email
-//       }
-//     });
-//   }
- };
+    const authtoken = jwt.sign({ id: uuid.v1(), }, process.env.MY_SECRET, {
+      expiresIn: "2d"
+    });
+    return res.status(201).json({
```

Comment:
Should we be returning 201 on a successful login? uuh!! check it up: return 200 instead

Reply:
I collected it